### PR TITLE
chore: shrink error messages shipped to client

### PR DESCRIPTION
**Pull request by benmccann**

### CODE REVIEW

File: packages/kit/src/utils/routing.js [hunk 1]
```diff
@@ -65,7 +67,9 @@ export function parse_route_id(id) {
 										const match = param_pattern.exec(content);
 										if (!match) {
```
Comment (gtm-nayan) [inline packages/kit/src/utils/routing.js:68]: Could remove this whole block on the client side I think, the manifest generation would fail on this so it never gets here on the client.

**Comment by Rich-Harris** [comment]
I think we should probably have a more principled approach to this problem — rather than making messages shorter (read: less helpful) in production, what if we did something similar to what React does and printed errors/warnings with codes that pointed to documentation? It could be as simple as [this](https://rollupjs.org/repl/?version=4.9.4&shareable=JTdCJTIyZXhhbXBsZSUyMiUzQSUyMiUyMiUyQyUyMm1vZHVsZXMlMjIlM0ElNUIlN0IlMjJjb2RlJTIyJTNBJTIyY29uc3QlMjBwcm9kJTIwJTNEJTIwdHJ1ZSUzQiU1Q24lNUNuZnVuY3Rpb24lMjB3YXJuKGNvZGUlMkMlMjBtZXNzYWdlKSUyMCU3QiU1Q24lMjAlMjBpZiUyMCghcHJvZCklMjBjb25zb2xlLmxvZyhtZXNzYWdlKSUzQiU1Q24lMjAlMjBjb25zb2xlLmxvZyglNjBodHRwcyUzQSUyRiUyRmtpdC5zdmVsdGUuZGV2JTJGZG9jcyUyRndhcm5pbmdzJTJGJTI0JTdCY29kZSU3RCU2MCklM0IlNUNuJTdEJTVDbiU1Q253YXJuKDEyMyUyQyUyMCdzb21ldGhpbmclMjB3ZW50JTIwd3JvbmcnKSUzQiUyMiUyQyUyMmlzRW50cnklMjIlM0F0cnVlJTJDJTIybmFtZSUyMiUzQSUyMm1haW4uanMlMjIlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyb3V0cHV0JTIyJTNBJTdCJTIyZm9ybWF0JTIyJTNBJTIyZXMlMjIlN0QlN0QlN0Q=)

File: packages/kit/src/utils/routing.js (hunk 1 above)
Comment (benmccann) [reply packages/kit/src/utils/routing.js]: done

**Comment by dummdidumm** [comment]
We definitly should have such an approach to our error messaging in the mid term

File: packages/kit/src/utils/routing.js [hunk 2]
```diff
@@ -63,11 +63,10 @@ export function parse_route_id(id) {
...
 										const match = param_pattern.exec(content);
-										if (!match) {
-											throw new Error(
-												`Invalid param: ${content}. Params and matcher names can only have underscores and alphanumeric characters.`
-											);
-										}
+										// Skip invalid params.
+										// Params and matcher names can only have underscores and alphanumeric characters.
+										// We'll never hit this because mnifest generation would have failed.
+										if (!match) return;
```
Comment (Rich-Harris) [inline packages/kit/src/utils/routing.js:69]: if we never hit it, why do we have this line at all? surely a type assertion would be sufficient?

we should probably be separating these changes into different PRs — this isn't related to error messages (which i think we should solve properly rather than with piecemeal changes), and the `disable_hash` stuff is a behaviour change that probably warrants discussion

Comment (benmccann) [reply packages/kit/src/utils/routing.js]: I like your idea about how to handle error messages. If we had that, I don't think we should change this though - better to ship no error message than a small unreachable one

**Comment by benmccann** [comment]
> I think we should probably have a more principled approach to this problem — rather than making messages shorter (read: less helpful) in production, what if we did something similar to what React does and printed errors/warnings with codes that pointed to documentation? It could be as simple as [this](https://rollupjs.org/repl/?version=4.9.4&shareable=JTdCJTIyZXhhbXBsZSUyMiUzQSUyMiUyMiUyQyUyMm1vZHVsZXMlMjIlM0ElNUIlN0IlMjJjb2RlJTIyJTNBJTIyY29uc3QlMjBwcm9kJTIwJTNEJTIwdHJ1ZSUzQiU1Q24lNUNuZnVuY3Rpb24lMjB3YXJuKGNvZGUlMkMlMjBtZXNzYWdlKSUyMCU3QiU1Q24lMjAlMjBpZiUyMCghcHJvZCklMjBjb25zb2xlLmxvZyhtZXNzYWdlKSUzQiU1Q24lMjAlMjBjb25zb2xlLmxvZyglNjBodHRwcyUzQSUyRiUyRmtpdC5zdmVsdGUuZGV2JTJGZG9jcyUyRndhcm5pbmdzJTJGJTI0JTdCY29kZSU3RCU2MCklM0IlNUNuJTdEJTVDbiU1Q253YXJuKDEyMyUyQyUyMCdzb21ldGhpbmclMjB3ZW50JTIwd3JvbmcnKSUzQiUyMiUyQyUyMmlzRW50cnklMjIlM0F0cnVlJTJDJTIybmFtZSUyMiUzQSUyMm1haW4uanMlMjIlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyb3V0cHV0JTIyJTNBJTdCJTIyZm9ybWF0JTIyJTNBJTIyZXMlMjIlN0QlN0QlN0Q=)

Yes, I like that idea. If we do that, I think I'd still want all the change here:
- we shouldn't ship an unreachable error
- removing the word "was" makes the error messages read better
- we're also removing code that we could avoid on the client and a smaller error message alone doesn't get us that

I could clarify the changeset message to say "remove and clarify error messages" or something along those lines. I'd be happy to break this into multiple PRs, but size-wise this is a nice size PR and we'd end up with three pretty small PRs if I did each of those changes separately. I do like the idea of having an errors page, but if we're wanting to split PRs, I don't think I'd mix it in with these changes and it would be better to do it separately.

**Review by Rich-Harris — APPROVED** [review]
